qutip/tests/test_superop_reps.py

```diff
@@ -148,204 +167,205 @@ def test_NeglectSmallKraus(self):
+                                to_super(qeye(dim2))),(1, 5), (2, 6))
 
-        for N in (2, 4, 8):
-            case(rand_super_bcsz(N))
+        A, B = to_stinespring(chan)
 
-    def test_chi_known(self):
+        assert A.dims == [[dim2, 1], [dim2]]
+        assert B.dims == [[dim1, 1], [dim1]]
+
+
+    @pytest.mark.parametrize('dimension', [2, 4, 8])
+    def test_chi_choi_roundtrip(self, dimension):
+
+        superop = rand_super_bcsz(dimension)
+        superop = to_chi(superop)
+        rt_superop = to_chi(to_choi(superop))
+        dif = norm(rt_superop.data.toarray() - superop.data.toarray())
+
+        assert 0 == pytest.approx(dif, abs=1e-7)
+        assert rt_superop.type == superop.type
+        assert rt_superop.dims == superop.dims
+
+    chi_sigmax = [
+        [0, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]
+    ]
+    chi_super_sigmax = [
+        [0, 0, 0, 0],
+        [0, 4, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]
+    ]
+    chi_diag2 = [
+        [4, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]
+    ]
+    rotX_pi_4 = (-1j * sigmax() * pi / 4).expm()
+    chi_rotX_pi_4 = [
+        [2, 2j, 0, 0],
+        [-2j, 2, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]
+    ]
+
+
+    @pytest.mark.parametrize(['superop', 'chi_expected'], 
+                             [pytest.param(sigmax(), chi_sigmax),
+                             pytest.param(to_super(sigmax()), chi_super_sigmax),
+                             pytest.param(qeye(2), chi_diag2),
+                             pytest.param(rotX_pi_4, chi_rotX_pi_4)
+                             ])
+    def test_chi_known(self, superop, chi_expected):
         """
         Superoperator: Chi-matrix for known cases is correct.
         """
-        def case(S, chi_expected, silent=True):
-            chi_actual = to_chi(S)
-            chiq = Qobj(chi_expected, dims=[[[2], [2]], [[2], [2]]], superrep='chi')
-            if not silent:
-                print(chi_actual)
-                print(chi_expected)
-            assert_almost_equal((chi_actual - chiq).norm('tr'), 0)
-
-        case(sigmax(), [
-            [0, 0, 0, 0],
-            [0, 4, 0, 0],
-            [0, 0, 0, 0],
-            [0, 0, 0, 0]
-        ])
-        case(to_super(sigmax()), [
-            [0, 0, 0, 0],
-            [0, 4, 0, 0],
-            [0, 0, 0, 0],
-            [0, 0, 0, 0]
-        ])
-        case(qeye(2), [
-            [4, 0, 0, 0],
-            [0, 0, 0, 0],
-            [0, 0, 0, 0],
-            [0, 0, 0, 0]
-        ])
-        case((-1j * sigmax() * pi / 4).expm(), [
-            [2, 2j, 0, 0],
-            [-2j, 2, 0, 0],
-            [0, 0, 0, 0],
-            [0, 0, 0, 0]
-        ])
+        chi_actual = to_chi(superop)
+        chiq = Qobj(chi_expected, dims=[[[2], [2]], [[2], [2]]], superrep='chi')
+        assert 0 == pytest.approx((chi_actual - chiq).norm('tr'), abs=1e-7)
```

Comment:
Two points:
1. an incredibly minor niggle, but I find these to be rather more readable if you put the constant inside the approximation, e.g. `(chi_actual - chiq).norm() == pytest.approx(0, abs=1e-7)`
2. `pytest.approx` isn't entirely approprioate here anyway - comparing a strictly non-negative result to 0 is rather clearer with `(chi_actual - chiq).norm() < tol`